pkg/provider/aws/resources/securitygroup.go

```diff
@@ -0,0 +1,46 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/klothoplatform/klotho/pkg/config"
+	"github.com/klothoplatform/klotho/pkg/core"
+)
+
+type SecurityGroup struct {
+	Name          string
+	Vpc           *Vpc
+	ConstructsRef []core.AnnotationKey
+}
+
+const SG_TYPE = "security_group"
+
+// GetSecurityGroup returns the security group if one exists, otherwise creates one, then returns it
+func GetSecurityGroup(cfg *config.Application, dag *core.ResourceGraph) *SecurityGroup {
+	for _, r := range dag.ListResources() {
+		if sg, ok := r.(*SecurityGroup); ok {
+			return sg
+		}
+	}
+	sg := &SecurityGroup{
+		Name: cfg.AppName,
+		Vpc:  GetVpc(cfg, dag),
+	}
+	dag.AddResource(sg)
```

Comment:
this has to have a dependency on the vpc and subnets likely because were going to create this all at one rather than doing the add rules and since we use the subnet cidr blocks we need the iac to come after those